### PR TITLE
Moved out logic from PrintModal to Print Helper

### PR DESCRIPTION
**Pull request by andreysaf**
PrintModal should have logic only for rendering the component, so made sense to move it out of PrintModal and into PrintHelper. 

Testing it on dev freezes Chrome due to react/chrome issue:
facebook/react: Issue #17155
facebook/react: Issue #16734

So to test it out, build the UI and place it in WebViewer 7.0.

Ran through all combinations and ensured nothing broke.

### CODE REVIEW

**Comment by andreysaf** [comment]
This PR is needed for me to introduce more options to trigger instance.print(..options) programmatically to include options like include annots, comments, page ranges etc.

File: src/components/PrintModal/PrintModal.js [hunk 1]
```diff
@@ -156,310 +153,24 @@ class PrintModal extends React.PureComponent {
       core.setWatermark(this.state.existingWatermarks);
     }
 
-    const creatingPages = this.creatingPages();
-    Promise.all(creatingPages)
+    const createPages = creatingPages(this.state.pagesToPrint, this.state.includeComments, this.state.includeAnnotations, this.props.printQuality, this.props.sortStrategy, this.props.colorMap);
+    Promise.all(createPages)
       .then(pages => {
-        this.printPages(pages);
+        debugger;
```
Comment (mparizeau-pdftron) [inline src/components/PrintModal/PrintModal.js:159]: Should remove debugger

Comment (andreysaf) [reply src/components/PrintModal/PrintModal.js]: Resolved in 45eb354 and rebased it.

**Comment by ZhijieZhang** [comment]
Removing myself as a reviewer since Matt is already taking a look

**Comment by ZhijieZhang** [comment]
Got a message from Xodo, adding myself back to one of the reviewers :^)

File: src/helpers/print.js [hunk 2]
```diff
@@ -53,3 +63,284 @@ const printPdf = () =>
...
+        printableAnnotations.forEach(annot => (annot.Printable = false));
+        await drawAnnotationsOnCanvas(canvas, pageNumber);
+        // enable all printable annotations after draw
+        printableAnnotations.forEach(annot => (annot.Printable = true));
+      }
+
+      const img = document.createElement('img');
+      img.src = canvas.toDataURL();
+      img.onload = () => {
+        count = count < 0 ? -1 : count + 1;
```
[the remark this answers is not in the document]
Comment (andreysaf) [reply src/helpers/print.js]: Resolved in 911e9dc.

File: src/components/PrintModal/PrintModal.js [hunk 3]
```diff
@@ -156,310 +153,23 @@ class PrintModal extends React.PureComponent {
       core.setWatermark(this.state.existingWatermarks);
     }
 
-    const creatingPages = this.creatingPages();
-    Promise.all(creatingPages)
+    const createPages = creatingPages(this.state.pagesToPrint, this.state.includeComments, this.state.includeAnnotations, this.props.printQuality, this.props.sortStrategy, this.props.colorMap);
```
[the remark this answers is not in the document]
Comment (andreysaf) [reply src/components/PrintModal/PrintModal.js]: resolved in 911e9dc.

**Comment by andreysaf** [comment]
print

**Comment by andreysaf** [comment]
Please do not delete branch after pulling it in, thanks 🙏

**Pull request merged**